### ISSUE TITLE
^Z issue

### ISSUE THREAD

**Issue by francoisvignon** [tbodt/ish#256]
A small ^Z issue:
if you key in 
```
sleep 3600 &[enter]
sleep 100[enter]
```
and then you key in `^Z`
nothing happen, you also key in `^C`
nothing happen

in fact all happen when the first task (sleep 3600 in this case) end.
you have a full sample bellow:
```
		iFv:~# sleep 21 &                                
		iFv:~# sleep 20                                  
		^Z^C^C                                           
		[2]+  Stopped                    sleep 20        
		[1]   Done                       sleep 21        
		iFv:~#                                           
```
all action occurs then sleep 21 end

you can also do by another way:
```
iFv:~# sleep 100                                 
^Z[1]+  Stopped                    sleep 100     
iFv:~# sleep 101                                 
^Z^Z^Z^Z^C                                       
```

and then wait 100 after you key in sleep 100 to recover usage :-)

please note also the following issue
```
iFv:~# sleep 100                                 
^Z[1]+  Stopped                    sleep 100          
iFv#: kill %1
iFv#: ps                                        
PID   USER     TIME  COMMAND                     
    1 root      0:00 /bin/login -f root          
    2 root      0:00 -ash                        
    6 root      0:00 sleep 100                   
    7 root      0:00 ps                          
iFv:~#        
```
nothing happen, to kill %1, you must backgrounding %1 (`bg %1`) then `kill %1`.
seem not allowed to kill a process in suspending state

it's not related to busybox (I try with another), not to ash, nor to bash (tried both)

**Comment by tbodt** [comment]
Fixed in 842da48

**Comment by tbodt** [comment]
The only signal that's supposed to have any effect on a stopped process is SIGKILL.

**Comment by francoisvignon** [comment]
maybe, but even this one don't work:

```
iFv:~# sleep 3600                                
^Z[1]+  Stopped                    sleep 3600    
iFv:~# kill -15 %1                               
iFv:~#                                           
iFv:~# ps                                        
PID   USER     TIME  COMMAND                     
    1 root      0:00 /bin/login -f root          
    2 root      0:00 -ash                        
   10 root      0:00 sleep 3600                  
   11 root      0:00 ps                          
iFv:~#                                           
iFv:~# kill -9 %1                                
iFv:~#                                           
iFv:~# ps                                        
PID   USER     TIME  COMMAND                     
    1 root      0:00 /bin/login -f root          
    2 root      0:00 -ash                        
   10 root      0:00 sleep 3600                  
   12 root      0:00 ps                          
iFv:~#
```

I have tried on some plateform (debian/MacOs) signal 9 (SIGKILL) and 15 (SIGTERM) seem working in suspended state.

SIGTERM seem awaken the task before terminating (because, previously, I send a Bus error signal, ignored in suspended state) and when I send SIGTERM, the Bus Error was displayed

**Comment by tbodt** [comment]
I just tried on Linux and SIGTERM didn't work. Not sure exactly how this is supposed to work then.

**Comment by francoisvignon** [comment]
from my Debian (with bash):
```
Linux mac32 4.9.0-3-686 #1 SMP Debian 4.9.30-2+deb9u5 (2017-09-19) i686 GNU/Linux
debian@mac32:~$ sleep 3600
^Z
[1]+  Stopped                 sleep 3600
debian@mac32:~$ ps
  PID TTY          TIME CMD
  476 pts/0    00:00:00 bash
  503 pts/0    00:00:00 sleep
  504 pts/0    00:00:00 ps
debian@mac32:~$ kill 503
debian@mac32:~$ ps
  PID TTY          TIME CMD
  476 pts/0    00:00:00 bash
  503 pts/0    00:00:00 sleep
  505 pts/0    00:00:00 ps
debian@mac32:~$
debian@mac32:~$ ps
  PID TTY          TIME CMD
  476 pts/0    00:00:00 bash
  503 pts/0    00:00:00 sleep
  506 pts/0    00:00:00 ps
debian@mac32:~$ kill %1

[1]+  Stopped                 sleep 3600
debian@mac32:~$ ps
  PID TTY          TIME CMD
  476 pts/0    00:00:00 bash
  507 pts/0    00:00:00 ps
[1]+  Terminated              sleep 3600
debian@mac32:~$
```

kill 'pid' don't work but kill %1 work.
I don't understand why it's a difference between kill 'pid' and kill %1 :-S, but kill %1 have a different cinematic than kill 'pid'

and now
```
debian@mac32:~$ sleep 3600
^Z
[1]+  Stopped                 sleep 3600
debian@mac32:~$ ps
  PID TTY          TIME CMD
  476 pts/0    00:00:00 bash
  508 pts/0    00:00:00 sleep
  509 pts/0    00:00:00 ps
debian@mac32:~$ kill -9 508
debian@mac32:~$
[1]+  Killed                  sleep 3600
debian@mac32:~$
```
kill -9 'pid' work directly

**Comment by tbodt** [comment]
Bash probably has a special case for kill %1 that sends a SIGCONT first. fish and busybox ash both don't do this.

**Comment by francoisvignon** [comment]
for %1, no time to debug bash ;-)

but for kill -9, I think this one should be made available in suspended state.
according some documentation:
```The signals SIGKILL and SIGSTOP cannot be caught or ignored.```

and in this case, you ignore SIGKILL and SIGSTOP in suspended state ;-)

**Comment by tbodt** [comment]
Should be fixed now (e5ca15aeb239fb34808ababa1ddac81da93d5882)

**Comment by francoisvignon** [comment]
Nice :-)

thanks for your reactivity
